Cannot find module `@sveltejs/kit/hooks` in dev server

### Describe the bug

When trying to use the newly merged `sequence` from `@sveltejs/kit/hooks` the dev server fails to load the module (and therefore fails to run any hooks).

Adapter-node, however, functions properly and chains the hooks (which also means it's not just a dependency installation issue).

### Reproduction

Reproduction repo: https://github.com/Karlinator/reproduction-sveltekit-hooks

As stated, this bug applies (as far as I can tell) only in dev mode. Haven't tested any other adapters than node, but that works.

### Logs

```shell
Error: Cannot find module '@sveltejs/kit/hooks' from '/home/karl/Git/reproduction-sveltekit-hooks/src'
    at Function.resolveSync [as sync] (/home/karl/Git/reproduction-sveltekit-hooks/node_modules/resolve/lib/sync.js:102:15)
    at resolveFrom$3 (/home/karl/Git/reproduction-sveltekit-hooks/node_modules/vite/dist/node/chunks/dep-98dbe93b.js:4079:29)
    at resolve (/home/karl/Git/reproduction-sveltekit-hooks/node_modules/vite/dist/node/chunks/dep-98dbe93b.js:73902:22)
    at nodeRequire (/home/karl/Git/reproduction-sveltekit-hooks/node_modules/vite/dist/node/chunks/dep-98dbe93b.js:73881:25)
    at ssrImport (/home/karl/Git/reproduction-sveltekit-hooks/node_modules/vite/dist/node/chunks/dep-98dbe93b.js:73823:20)
    at eval (//home/karl/Git/reproduction-sveltekit-hooks/src/hooks.ts:3:37)
    at instantiateModule (/home/karl/Git/reproduction-sveltekit-hooks/node_modules/vite/dist/node/chunks/dep-98dbe93b.js:73866:15)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```


### System Info

```shell
System:
    OS: Linux 5.13 Fedora 34 (Workstation Edition) 34 (Workstation Edition)
    CPU: (8) x64 Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
    Memory: 656.26 MB / 15.37 GB
    Container: Yes
    Shell: 5.1.0 - /bin/bash
  Binaries:
    Node: 14.17.0 - /usr/bin/node
    npm: 7.20.3 - /usr/local/bin/npm
  Browsers:
    Firefox: 90.0.2
  npmPackages:
    @sveltejs/adapter-node: ^1.0.0-next.39 => 1.0.0-next.42 
    @sveltejs/kit: next => 1.0.0-next.153 
    svelte: ^3.34.0 => 3.42.1
```


### Severity

serious, but I can work around it

### Additional Information

Currently working around by either compiling with adapter-node during development or just not using `sequence`.

--- a/.changeset/short-emus-juggle.md
+++ b/.changeset/short-emus-juggle.md
@@ -1,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+Add `@sveltejs/kit` to noExternal to resolve hooks module in dev server

--- a/packages/kit/src/core/build/index.js
+++ b/packages/kit/src/core/build/index.js
@@ -488,10 +488,12 @@ async function build_server(
 			}
 		},
 		ssr: {
-			// note to self: this _might_ need to be ['svelte', '@sveltejs/kit', ...get_no_external()]
-			// but I'm honestly not sure. roll with this for now and see if it's ok
-			// @ts-expect-error - ssr is considered in alpha, so not yet exposed by Vite
-			noExternal: [...((vite_config.ssr && vite_config.ssr.noExternal) || []), ...svelte_packages]
+			noExternal: [
+				'@sveltejs/kit',
+				// @ts-expect-error - ssr is considered in alpha, so not yet exposed by Vite
+				...((vite_config.ssr && vite_config.ssr.noExternal) || []),
+				...svelte_packages
+			]
 		}
 	});
 

--- a/packages/kit/src/core/build/index.js
+++ b/packages/kit/src/core/build/index.js
@@ -489,7 +489,7 @@ async function build_server(
 		},
 		ssr: {
 			noExternal: [
-				'@sveltejs/kit',
+				'@sveltejs/kit', // TODO: see https://github.com/vitejs/vite/issues/3953
 				// @ts-expect-error - ssr is considered in alpha, so not yet exposed by Vite
 				...((vite_config.ssr && vite_config.ssr.noExternal) || []),
 				...svelte_packages

